master/internal/api_experiment.go

```diff
@@ -1916,3 +1923,158 @@ func (a *apiServer) GetModelDefFile(
+		Table("ex").
+		ColumnExpr("ex.config->'searcher'->>'metric' AS metric_name").
+		ColumnExpr("(SELECT CASE WHEN coalesce((ex.config->'searcher'->>'smaller_is_better')::boolean, true) THEN 1 ELSE -1 END) as sign").
+		ColumnExpr("trials.id AS trial_id").
+		Join("JOIN trials ON trials.experiment_id = ex.id")
+
+	// get info for best/latest validation for best trial
+	validationsQuery := db.Bun().NewSelect().
+		Table("validations").
+		Column("id").
+		ColumnExpr("validations.trial_id as trial_id").
+		Column("total_batches").
+		ColumnExpr("proto_time(end_time) AS end_time").
+		ColumnExpr("json_build_object('avg_metrics', metrics->'validation_metrics') AS metrics").
+		ColumnExpr("metrics->'num_inputs' AS num_inputs").
+		ColumnExpr("((metrics->'validation_metrics'->>(si.metric_name))::float8 * si.sign) AS signed_searcher_metric").
+		ColumnExpr("row_number() OVER(PARTITION BY validations.trial_id ORDER BY end_time DESC NULLS LAST) AS latest_rank").
+		Join("JOIN si ON validations.trial_id = si.trial_id")
+
+	// get best checkpoint info for best trial
+	addCheckpointColumnsAndJoin := func(q *bun.SelectQuery) *bun.SelectQuery {
+		return q.
+			Column("v.signed_searcher_metric").
+			ColumnExpr("c.*").
+			ColumnExpr("row_number() OVER(PARTITION BY v.trial_id ORDER BY v.signed_searcher_metric ASC) AS rank").
+			Join("JOIN v ON c.steps_completed = v.total_batches AND c.trial_id = v.trial_id").
+			Where("c.state = 'COMPLETED'")
+	}
+
+	oldCheckpointInnerQuery := db.Bun().NewSelect().
+		TableExpr("checkpoints_old_view c").
+		Apply(addCheckpointColumnsAndJoin)
+
+	oldCheckpointOuterQuery := db.Bun().NewSelect().
+		TableExpr("(?) as old_c", oldCheckpointInnerQuery).
+		ColumnExpr("old_c.*").
+		Where("old_c.rank = 1")
+
+	newCheckpointInnerQuery := db.Bun().NewSelect().
+		TableExpr("checkpoints_new_view c").
+		Apply(addCheckpointColumnsAndJoin)
+
+	newCheckpointOuterQuery := db.Bun().NewSelect().
+		TableExpr("(?) as new_c", newCheckpointInnerQuery).
+		ColumnExpr("new_c.*").
+		Where("new_c.rank = 1")
+
+	checkpointInnerQueryUnion := oldCheckpointOuterQuery.UnionAll(newCheckpointOuterQuery)
+
+	checkpointInnerQuery := db.Bun().NewSelect().
+		TableExpr("(?) as bc", checkpointInnerQueryUnion).
+		ColumnExpr("bc.*").
+		ColumnExpr("row_number() OVER(PARTITION BY bc.trial_id ORDER BY bc.signed_searcher_metric) as order_rank")
+
+	checkpointsQuery := db.Bun().NewSelect().
+		TableExpr("(?) as c", checkpointInnerQuery).
+		Column("c.trial_id").
+		Column("c.resources").
+		ColumnExpr("c.order_rank AS rank").
+		ColumnExpr("c.uuid::text AS uuid").
+		ColumnExpr("c.steps_completed AS total_batches").
+		ColumnExpr("proto_time(c.report_time) AS end_time").
+		ColumnExpr(protoStateDBCaseString(checkpointv1.State_value, "c.state", "state", "STATE_"))
+
+	stepsQuery := db.Bun().NewSelect().
+		TableExpr("steps AS s").
+		Column("s.total_batches").
+		Where("s.trial_id = trials.id").
+		Order("s.total_batches DESC").
+		Limit(1)
```

Comment:
do we need this? i could see it being useful, but joins involving steps have had performance issues in the past and i didn't see it in the doc

Reply:
The query fails to scan if the column is missing. We could pass null in the column, but returning the same type with fields missing based on the endpoint makes my teeth itch a little.

Reply:
this subquery can be eliminated after https://hpe-aiatscale.atlassian.net/browse/DET-9154

Reply:
Let's add that as a TODO since this is feature-switch gated and blocking the frontend portion.